Tevling/Components/ChallengeCard.razor.js

```diff
@@ -0,0 +1,21 @@
+﻿window.DrawChallengeStats = function (canvasId, data, labels, chartTitle) {
+    var ctx = document.getElementById(canvasId).getContext('2d');
+    var canvas = document.getElementById(canvasId);
+
+    // Destroy existing chart instance if it exists
+    if (window.canvasId instanceof Chart) {
```

Comment:
This should be `window[canvasId]` to get the correct object. Same below where it's defined.